File: 2D-World-Simulator-Java/OPP-Java/src/op/JavaProject/Functionality/BoardPoint.java
package op.JavaProject.Functionality;

import op.JavaProject.Game.World;

import java.io.Serializable;
import java.util.Random;

public class BoardPoint implements Serializable {
    private int x;
    private int y;

    //default constructor
    public BoardPoint() {
        this.x = 0;
        this.y = 0;
    }

    //constructor with two parameters
    public BoardPoint(int x, int y) {
        this.x = x;
        this.y = y;
    }

    //getting random point in the range of world size
    public void SetRandomPoint(World organismWorld) {
        int randX = GetRandomValue(0, organismWorld.GetWidth());
        int randY = GetRandomValue(0, organismWorld.GetHeight());
        SetX(randX);
        SetY(randY);
    }

    //getting random value
    public static int GetRandomValue(int minimumValue, int maximumValue) {
        Random randomValue = new Random();
        return randomValue.nextInt(maximumValue - minimumValue) + minimumValue;
    }

    public int GetX() {
        return this.x;
    }

    public int GetY() {
        return this.y;
    }

    public void SetX(int x) {
        this.x = x;
    }

    public void SetY(int y) {
        this.y = y;
    }

    //function to compare instances of points
    @Override
    public boolean equals(Object obj) {
        if (obj instanceof BoardPoint) {
            BoardPoint other = (BoardPoint) obj;
            return (x == other.x && y == other.y);
        }
        return false;
    }

}

File: 2D-World-Simulator-Python/OPP-Python/Functionality/BoardPoint.py
import random


class BoardPoint:

    def __init__(self, x, y):
        self.x = x
        self.y = y

    # getting random point in range of world size
    def set_random_point(self, organism_world):
        self.x = random.randint(0, organism_world.width - 1)
        self.y = random.randint(0, organism_world.height - 1)

    def __eq__(self, other):
        if isinstance(other, BoardPoint):
            return self.x == other.x and self.y == other.y
        return False

    def __ne__(self, other):
        return not self.__eq__(other)

    def get_x(self):
        return self.x

    def get_y(self):
        return self.y

    def set_x(self, new_x):
        self.x = new_x

    def set_y(self, new_y):
        self.y = new_y
